hyperledger/iroha
# fix(snapshot): verify all blocks while reading snapshot

## Description

Changes:
- relative path to snapshot folder is resolved in the same way as for `kura`
- last block hash check of snapshot was missed before:
    - not it is checked
    - if check fail last changes to the state are discarded without losing the whole snapshot







### Benefits

- Behavior of configuration is consistent 
- All blocks are checked

### How to test

1. Run iroha
2. Wait until iroha create snapshot
3. Stop iroha
4. Modify last hash in `block_hashes`.`blocks` (any character except for last one)
5. Load iroha
6. Verify that iroha is loaded from snapshot
7. Check that last block was discarded (in logs)
8. Check that iroha can continue to commit new blocks



### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up
